docs/directindex.adoc

```diff
@@ -59,6 +59,10 @@ Names of property keys that contain non-alphanumeric characters must be placed i
 [source, java]
 graph.indexQuery("vertexByText", "v.\"first_name\":john").vertices()
 
+Some property key names may be transformed by the JanusGraph indexing backend implementation. For instance, an indexing backend that does not permit spaces in field names may transform "My Field Name" to "My•Field•Name", or an indexing backend like Solr may append type information to the name, transforming "myBooleanField" to "myBooleanField_b". These transformations happen in the index backend's implementation of IndexProvider, in the "mapKey2Field" method. Indexing backends may reserve special characters (such as '•') and prohibit indexing of fields that contain them.
+
+In general, making direct index queries depends on implementation details of JanusGraph indexing backends that are normally hidden from users, so it's best to verify a query empirically against the indexing backend in use.
```

Comment:
It's probably worthwhile to make a general recommendation in the docs: If possible, only use alphanumeric and the underscore for property names.

This might be more appropriate/visible in the [schema and data modeling](https://github.com/JanusGraph/janusgraph/blob/master/docs/basics.adoc#defining-property-keys) docs.

Reply:
@sharpau any updates on this point?

Reply:
Added a recommendation here and in basics.adoc. I'm going with that in lieu of making it configurable, as it's now probably obvious that I'd just like to get this finished. 😄